core/engine/src/value/conversions/try_from_js.rs

```diff
@@ -58,6 +59,33 @@ where
     }
 }
 
+impl<T> TryFromJs for Vec<T>
+where
+    T: TryFromJs,
+{
+    fn try_from_js(value: &JsValue, context: &mut Context) -> JsResult<Self> {
+        let object = match value {
+            JsValue::Object(o) => o,
+            _ => {
+                return Err(JsNativeError::typ()
+                    .with_message("cannot convert value to a Vec")
+                    .into())
+            }
+        };
+
+        let length = object
+            .get(js_string!("length"), context)?
+            .to_length(context)?;
+        let mut vec = Vec::with_capacity(length as usize);
```

Comment:
I'd avoid casting to `usize` and use `usize::try_from` with some added context in case the conversion fails.

Reply:
Done.